dlc-messages/src/ser_impls.rs

```diff
@@ -453,7 +453,14 @@ pub fn read_option_cb<R: ::std::io::Read, T, F>(
 where
     F: Fn(&mut R) -> Result<T, DecodeError>,
 {
-    let prefix: u8 = Readable::read(reader)?;
+    let prefix: u8 = match Readable::read(reader) {
```

Comment:
I wonder if it'd be better to have a different option type that allows the field not being present?

Reply:
Maybe! I thought this patch wouldn't be controversial, but it's fair to say that we don't need it since we dropped the other one. I'll remove this one too!